### ISSUE TITLE
How to verify if YOLOv8 custom model successfully detects custom object?

### ISSUE THREAD

**Issue by ZephTheHungryLoli** [ultralytics/ultralytics#1746]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to write a function that is only called when YOLOv8 object detection is successful.

So how can I verify whether the model detects the custom object or not?

Is there any function in YOLOv8 that returns a boolean value of true if object detection is successful?

**Comment by glenn-jocher** [comment]
@ZephTheHungryLoli yes, this is the default use case. See https://docs.ultralytics.com/modes/predict/ for handling prediction results.

**Comment by ZephTheHungryLoli** [comment]
@glenn-jocher I understand that, but how can I know if the model is detecting successfully or not? In the documentation, it's written that:

> The Results object contains the following components:

    Results.boxes: Boxes object with properties and methods for manipulating bounding boxes
    Results.masks: Masks object for indexing masks or getting segment coordinates
    Results.probs: torch.Tensor containing class probabilities or logits
    Results.orig_img: Original image loaded in memory
    Results.path: Path containing the path to the input image

So, I mean, is there any method that returns 'True' or 'False'  if the model detection IS SUCCESSFUL ?

**Comment by ahmedmuzammilAI** [comment]
> So, I mean, is there any method that returns 'True' or 'False' if the model detection IS SUCCESSFUL ?
@ZephTheHungryLoli 
Here's how I did it, this might help you out:
```
res = model.predict(image, conf=conf)
for result in res:
                boxes = result.boxes
                if len(boxes)==0:
                    print("no detections")
```

**Comment by glenn-jocher** [comment]
Hello @ahmedmuzammilAI 👋,

Regarding your question on verifying if the YOLOv8 model successfully detects custom objects, there is no method that directly returns a boolean value of `True` or `False` that indicates whether the object detection is successful or not.

However, you can use the `Results` object returned by the `predict()` function and check the length of the `boxes` object. If the length of `boxes` is zero, it means that no objects were detected in the image, and if it is greater than zero, it means that the model has detected custom objects. 

So, in your function, you can check the length of the `boxes` object and call the required function based on the return value.

Thank you for using YOLOv8. Let us know if you have any other questions!

**Comment by ZephTheHungryLoli** [comment]
Thank you for the clarification.

**Comment by glenn-jocher** [comment]
Hello @ZephTheHungryLoli 👋,

Glad to be of help! To verify whether the YOLOv8 model successfully detects custom objects, you can use the `Results` object returned by the `predict()` function. This object contains `boxes`, `masks`, `probs`, and other components.

You can check the length of the `boxes` object. If the length of `boxes` is zero, it means that no objects were detected in the image. If it is greater than zero, it means that the model has detected custom objects successfully. You can then call the required function based on the return value.

I hope this clarifies your question. Feel free to let us know if you have any other questions.

Thank you for using YOLOv8.

**Comment by ZephTheHungryLoli** [comment]
Awesome, thanks for the info!

**Comment by glenn-jocher** [comment]
Hello @ZephTheHungryLoli 👋,

To verify if custom objects are detected successfully by the YOLOv8 model, use the `Results` object returned by the `predict()` function. This object includes details such as `boxes`, `masks`, `probs`, and more.

You can check the length of the `boxes` object to see if custom objects are detected. If the length is zero, it means that no objects were detected in the image. If the length is greater than zero, it means that the model has detected custom objects.

I hope this helps in addressing your question. Please don't hesitate to let us know if you have any other queries.

Thank you for using YOLOv8.

**Comment by Baro1502** [comment]
Hi glenn, I am using YOLOv8 for detecting binary class. After training it generated one precision-recall curve but it seems "weird" - according to my boss. So we are asked to do it manually.

I am trying to evaluating it with each confidence value with the following code:
```
for conf in np.arange(.005,1.0001,.005):
    results = model.predict(source=test_data_folder,save_txt=True,save=True,conf=conf,stream = True)
```

How do I calculate precision-recall curve manually for the whole data folder, with each confidence/IoU point? And if I want to have annotations for all confidence detected to calculate mAP as [https://learnopencv.com/mean-average-precision-map-object-detection-model-evaluation-metric/](url) ?

Thank you

**Comment by pderrenger** [comment]
@Baro1502 hi Glenn, I am using YOLOv8 for binary object detection. After training, the generated precision-recall curve seems "weird" according to feedback from my boss. Thus, we are planning to perform the evaluation manually.

I am attempting to evaluate the model using different confidence values with the following code:
```python
for conf in np.arange(.005, 1.0001, .005):
    results = model.predict(source=test_data_folder, save_txt=True, save=True, conf=conf, stream=True)
```

I would like to know how to calculate the precision-recall curve manually for the entire data folder, considering each confidence/IoU point. Additionally, I want to acquire annotations for all confidence-detected items to calculate mAP as described in this [link](https://learnopencv.com/mean-average-precision-map-object-detection-model-evaluation-metric/).

Thank you
